### PR TITLE
Disable certain CPython tests on Windows

### PR DESCRIPTION
**Pull request by BasixKOR**
This patch allows running CPython tests fully on Windows. It doesn't fix failing tests yet.

### CODE REVIEW

File: Lib/test/test_exceptions.py [hunk 1]
```diff
@@ -1187,7 +1190,8 @@ def inner():
         else:
             self.fail("RecursionError not raised")
         self.assertEqual(wr(), None)
-
+    
+    @unittest.skipIf(sys.platform == 'win32', 'error specific to cpython')
```
Comment (youknowone) [inline Lib/test/test_exceptions.py:1194]: What do you mean by "error specific to cpython"?

Comment (BasixKOR) [reply Lib/test/test_exceptions.py:1194]: > Issue #12802: "not a directory" errors are ENOTDIR even on Windows 

It seems like CPython has a bug that errno returns `ENOTDIR` on Windows.

**Comment by BasixKOR** [comment]
Yeah, sily mistake 😅 I'll fix it when I rebase it to master.

**Comment by BasixKOR** [comment]
Converting to a draft for `test_os.py`

**Comment by BasixKOR** [comment]
test_os got fixed by #2095

**Pull request merged**